### ISSUE TITLE
Multisite root-implicit engine inherits the #778 rank clamp but keeps the pre-#778 1e-6 gate, so it raises on any physical state

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#784]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

## Summary

The multisite root-implicit engine inherits #778's rank clamp but not the gate
that has to go with it, so it raises `RootResidualError` on any physical state —
the exact failure #772 describes, on a code path #772's fix does not reach.

`_ctm_root_implicit_multisite.py` imports `_rank_capped_spectrum` from the asym
module and calls it with the derived `eps**(1/3)` clamp:

```python
# src/tenax/algorithms/_ctm_root_implicit_multisite.py:345-360
from tenax.algorithms._ctm_root_implicit_asym import (
    _inv_sqrt,
    _pin_bond_gauge,
    _rank_capped_spectrum,
)
...
    # Cap the usable rank and clamp the numerically-null tail (#772). The
    # same equations run here, so the same precision limit applies: ...
    s_k, _usable_rank = _rank_capped_spectrum(s, chi)
```

Its own comment is right — the same precision limit applies, so the same
intrinsic residual floor appears. But its gate is still the pre-#778 constant:

```python
# src/tenax/algorithms/_ctm_root_implicit_multisite.py:939-940
    root_residual_warn: float = 1e-6,
    on_root_residual: str = "raise",
```

Clamping the numerically-null tail leaves an O(`rel_floor`) inconsistency in the
characteristic equations *by construction*. A fixed `1e-6` therefore rejects
exactly the states the clamp has just made healthy.

## Measured

Frozen physical D=2 Heisenberg simple-update state (`tests/_su_fixtures.py`,
`physical_su_d2()`), 1×1 cell, `max_iter=300, conv_tol=1e-13`, default clamp:

| state | χ | gated `root_residual` | vs the `1e-6` default | gradient |
|---|---|---|---|---|
| physical | 4 | **8.493e-06** | **8.5× over → raises** | finite |
| physical | 6 | **1.911e-05** | **19× over → raises** | finite |
| random | 4 | 2.418e-14 | passes | finite |
| random | 6 | 2.397e-14 | passes | finite |

Two things this shows:

1. **The gradients are finite.** The clamp is doing its job here; multisite does
   not have the NaN half of #772. What it has is only the gate — i.e. precisely
   the state the asym engine was left in *after* #778 and *before* the gate was
   recalibrated.
2. **The floor is the same floor.** These values are numerically identical to the
   asym engine's *covariant* residual at the same χ (8.493e-06 at χ=4,
   1.911e-05 at χ=6, per the measurements in the #772 close-out). This is the
   same phenomenon, not a coincidence of magnitude.

## Why no test catches it

Every multisite test builds its cell from the random `_site_tensor` helper, whose
environment never approaches the clamp — its residual is ~2.4e-14, eight orders
inside the gate. This is the same blind spot that let #772 reach the wiring
stage: the contrast only appears on a physical state, and until recently the repo
had no physical-state fixture at all. `tests/_su_fixtures.py` now provides one.

## Repro

From a worktree with the `_su_fixtures` fixture available:

```python
import jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)
import tenax.algorithms._ctm_root_implicit_multisite as MS
from tests._su_fixtures import physical_su_d2

# Defaults raise; disable the gate only to read the number it would have rejected.
_E, grad, diag = MS.cell_root_implicit_energy_and_grad(
    {(0, 0): physical_su_d2()}, jnp.array([[0.5, 0.0], [0.0, -0.5]]),
    chi=4, nrows=1, ncols=1, max_iter=300, conv_tol=1e-13,
    on_root_residual="warn", root_residual_warn=float("inf"),
    return_diagnostics=True,
)
print(diag["root_residual"])            # 8.493e-06  -> would raise against 1e-6
print(all(bool(jnp.all(jnp.isfinite(g))) for g in grad.values()))  # True
```

Dropping `on_root_residual`/`root_residual_warn` (i.e. using the defaults) raises
`RootResidualError`.

## Suggested fix

The asym engine's close-out for #772 introduces
`_default_root_residual_warn(rel_floor) -> max(1e-6, 100 * rel_floor)`, which ties
the gate to the clamp that produces the residual, so lowering `rel_floor` restores
the tight gate where residuals genuinely are ~1e-13. Multisite should adopt the
same helper rather than growing a second, independently-tuned constant.

Two related loose ends on this path:

- **`rel_floor` is unreachable from multisite.** The call is
  `_rank_capped_spectrum(s, chi)` with no `rel_floor` argument, so the knob the
  asym engine now threads end-to-end cannot be set here at all. Whatever gate is
  chosen, it should be derivable from the clamp actually in force.
- **`covariant_residual` is computed but never gated.** It is placed in the
  diagnostics dict (line ~1083), but no `covariant_residual > ...` check exists in
  this module — unlike the asym and symmetric engines, which both gate it. At 1×1
  it equals `root_residual` (measured above), so nothing is lost there; whether
  that still holds on a genuine multi-site cell is **unverified**.

## Scope note

This was found while closing #772 and deliberately left out of that PR, whose
stated constraint is not to change `_ctm_root_implicit_multisite.py`'s behaviour.
Filing separately so #772 can close on the asym engine without the defect going
unrecorded.

Refs #772, #778, #779, #715.

**Comment by yingjerkao** [comment]
> 🤖 **AI-generated comment** — written by Claude Code, posted by @yingjerkao.

**Severity correction: this is latent, not live.**

The measurements above stand — calling `cell_root_implicit_energy_and_grad`
directly on a physical state does raise, and the residuals are as tabulated. But
the engine is **not reachable from the production wiring**, which I should have
checked before filing:

```python
# src/tenax/algorithms/ipeps_optimize_root_implicit.py:197-206
    if variant == "cell":
        raise NotImplementedError(
            "ctm_ad_mode='root_implicit' currently wires the dense 1x1 "
            "(asymmetric) engine only; got unit_cell=..."
            "The multisite engine (cell_root_implicit_energy_and_grad, #715 "
            "Phase 2) is built and tested but not yet wired here ..."
        )
```

So no user reaches this through `ctm_ad_mode="root_implicit"` today; it is
reachable only by calling the engine directly, as the repro does.

**Why it is still worth fixing rather than closing:** the defect is positioned
exactly where #715 Phase 2 will land. Wiring the multisite engine will hit this
on the first physical state it is pointed at, and the symptom — a hard
`RootResidualError` on a state whose gradients are actually fine — is easy to
misread as a bug in the new cell-shift index tables, which that phase's own
notes call "the whole risk of that phase". Fixing the gate first removes a
false lead from that debugging path.

Suggested sequencing: fold this into #715 Phase 2 as a precondition rather than
carrying it as an independent bug. The fix is small — adopt
`_default_root_residual_warn` and thread `rel_floor` to
`_rank_capped_spectrum(s, chi)` at line 360.
